agent_app_sample_code/agents/function_calling_agent_w_retriever_tool.py

```diff
@@ -0,0 +1,496 @@
+# Databricks notebook source
+# MAGIC %md
+# MAGIC # Function Calling Agent w/ Retriever 
+# MAGIC
+# MAGIC In this notebook, we construct the Agent with a Retriever tool.
+
+# COMMAND ----------
+
+# If running this notebook by itself, uncomment these.
+# %pip install --upgrade -qqqq databricks-agents databricks-vectorsearch
+# dbutils.library.restartPython()
+
+# COMMAND ----------
+
+import json
+import os
+from typing import Any, Callable, Dict, List, Optional, Union
+import mlflow
+from dataclasses import asdict, dataclass
+import pandas as pd
+from mlflow.models import set_model, ModelConfig
+from mlflow.models.rag_signatures import StringResponse, ChatCompletionRequest, Message
+from mlflow.deployments import get_deploy_client
+
+import os
+
+# COMMAND ----------
+
+# MAGIC %md ##### Retriever tool
+
+# COMMAND ----------
+
+from databricks.vector_search.client import VectorSearchClient
+
+
+@dataclass
+class Document:
+    page_content: str
+    metadata: Dict[str, str]
+    type: str
+
+
+class VectorSearchRetriever:
+    """
+    Class using Databricks Vector Search to retrieve relevant documents.
+    """
+
+    def __init__(self, config: Dict[str, Any]):
+        # TODO: Validate config
```

Comment:
Can this TODO be addressed (just looking through all of them)?

Reply:
removed. i dont believe config should be validated inside the agent code